contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,902 @@
+    require(_vote <= 1, "voting-rep-bad-vote");
+    require(getMotionState(_motionId) == MotionState.Staking, "voting-rep-staking-closed");
+
+    uint256 requiredStake = getRequiredStake(_motionId);
+    uint256 amount = min(_amount, sub(requiredStake, motion.stakes[_vote]));
+    uint256 stakerTotalAmount = add(stakes[_motionId][msg.sender][_vote], amount);
+
+    require(amount > 0, "voting-rep-bad-amount");
+
+    require(
+      stakerTotalAmount <= getReputationFromProof(_motionId, msg.sender, _key, _value, _branchMask, _siblings),
+      "voting-rep-insufficient-rep"
+    );
+    require(
+      stakerTotalAmount >= wmul(requiredStake, userMinStakeFraction) ||
+      add(motion.stakes[_vote], amount) == requiredStake, // To prevent a residual stake from being un-stakable
+      "voting-rep-insufficient-stake"
+    );
+
+    colony.obligateStake(msg.sender, motion.domainId, amount);
+    colony.transferStake(_permissionDomainId, _childSkillIndex, address(this), msg.sender, motion.domainId, amount, address(this));
+
+    // Update the stake
+    motion.stakes[_vote] = add(motion.stakes[_vote], amount);
+    stakes[_motionId][msg.sender][_vote] = stakerTotalAmount;
+
+    // Increment counter & extend claim delay if staking for an expenditure state change
+    if (
+      _vote == YAY &&
+      motion.stakes[YAY] == requiredStake &&
+      getSig(motion.action) == CHANGE_FUNCTION &&
+      add(motion.votes[NAY], motion.votes[YAY]) == 0
+    ) {
+      bytes32 structHash = hashExpenditureStruct(motion.action);
+      expenditureMotionCounts[structHash] = add(expenditureMotionCounts[structHash], 1);
+      bytes memory claimDelayAction = createClaimDelayAction(motion.action, UINT256_MAX);
+      require(executeCall(_motionId, claimDelayAction), "voting-rep-expenditure-lock-failed");
+    }
+
+    // Move to second staking window once one side is fully staked
+    if (
+      (_vote == YAY && motion.stakes[YAY] == requiredStake && motion.stakes[NAY] < requiredStake) ||
+      (_vote == NAY && motion.stakes[NAY] == requiredStake && motion.stakes[YAY] < requiredStake)
+    ) {
+      motion.events[STAKE_END] = uint64(now + stakePeriod);
+      motion.events[SUBMIT_END] = uint64(now + stakePeriod + submitPeriod);
+      motion.events[REVEAL_END] = uint64(now + stakePeriod + submitPeriod + revealPeriod);
+      delete motion.votes; // New stake supersedes prior votes
+
+      emit MotionEventSet(_motionId, STAKE_END);
+    }
+
+    // Claim tokens once both sides are fully staked
+    if (motion.stakes[YAY] == requiredStake && motion.stakes[NAY] == requiredStake) {
+      motion.events[STAKE_END] = uint64(now);
+      motion.events[SUBMIT_END] = uint64(now + submitPeriod);
+      motion.events[REVEAL_END] = uint64(now + submitPeriod + revealPeriod);
+      tokenLocking.claim(token, true);
+
+      emit MotionEventSet(_motionId, STAKE_END);
+    }
+
+    emit MotionStaked(_motionId, msg.sender, _vote, amount);
```

Comment:
Ah yes you're right, I missed that one. `MotionEventSet` is meant to indicate lifecycle events. For example, while `MotionStaked` indicates that a single user submitted a stake, it tells us nothing about the motion lifecycle (it could be the first stake, last stake, or anything in between). On the other hand, `MotionEventSet` fires only when we've hit a lifecycle event, such as a full stake or hitting our pseudo-quorum threshold in vote submissions. We don't emit it during escalation because `MotionEscalated` is already a lifecycle event.